library-valcon-internship/src/components/BookFormWrapper/BookFormWrapper.tsx

```diff
@@ -0,0 +1,268 @@
+import { Dispatch, SetStateAction, useEffect, useState } from 'react'
+
+import { useNavigate } from 'react-router-dom'
+
+import AuthorFormType from '../../models/AuthorFormType'
+import AuthorFormValidation from '../../models/AuthorFormValidation'
+import Book from '../../models/Book'
+import BookFormType from '../../models/BookFormType'
+import BookFormValidation from '../../models/BookFormValidation'
+import { addNewAuthor, getAllAuthors } from '../../services/AuthorService'
+import { addNewBook, editBook } from '../../services/BookService'
+import { BASE_64_EXTENSION, convertBase64ToBlob, initBookForm } from '../../utils/Utils'
+import AuthorForm from '../AuthorForm/AuthorForm'
+import BookForm from '../BookForm/BookForm'
+import Modal from '../Modal/Modal'
+import './BookFormWrapper.css'
+
+interface BookFormWrapperProps {
+  closeModal?: () => void
+  book?: Book
+  retrieveBook?: boolean
+  setRetrieveBook?: Dispatch<SetStateAction<boolean>>
+}
+
+const BookFormWrapper = ({ closeModal, book, retrieveBook,  setRetrieveBook }: BookFormWrapperProps) => {
+  const [ bookForm, setBookForm ] = useState<BookFormType>(initBookForm(book))
+  const [ bookFormValidation, setBookFormValidation ] = useState<BookFormValidation>({
```

Comment:
Default object value is looking a bit long to me. Crete object with these default values and assign it as initial state.